### DISCUSSION TITLE
Drivebase with Technics Hub

### DISCUSSION THREAD

**Discussion by Biggles765** [pybricks/support#1657] [Q&A]
I have done a test program while learning Pybricks and found that if I use drive_base forever or in python drive_base.drive it doesn't work but if I use drive_base.straight it does work. It appears to be the 'drive' function that isn't working for me.

**Comment by dlech** [comment]
Can you share the full program that you tried?

Some drive functions work in the background, so if you don't have a `wait(5000)` after the drive function, your program might reach the end of the file and stop. So it seems like the drive function isn't working, but actually, the program is just ended.

**Comment by Biggles765** [comment]
Thanks for that.
